### PR TITLE
fix: UserSessionChanged should be raised regardless of sid claim. Sid…

### LINKED ISSUE #1242: Session change not fired if identity token does not contain a sid claim
See here: https://github.com/authts/oidc-client-ts/blob/b7266eb650391d9e9807af21164874889963f1be/src/SessionMonitor.ts#L156

If there is no `sid` claim in the id token then `undefined === undefined` evaluates to true and the userSessionChanged event is never fired.

The `sid` claim is not part of the [session management spec](https://openid.net/specs/openid-connect-session-1_0.html), it is only defined in [frontchannel logout spec](https://openid.net/specs/openid-connect-frontchannel-1_0.html#OPLogout) and it's optional. This claim should not be checked at all or at least the user session should be assumed to have changed when the claim is not present.

### PR DESCRIPTION
**Pull request by bachratyg**
… is not part of the session management spec.

Closes/fixes #1242

### Checklist

- [X] This PR makes changes to the public API 
- [X] I have included links for closing relevant issue numbers

### CODE REVIEW

File: src/SessionMonitor.ts [hunk 1]
```diff
@@ -153,13 +153,8 @@ export class SessionMonitor {
                     raiseEvent = false;
                     this._checkSessionIFrame.start(session.session_state);
 
-                    if (session.sid === this._sid) {
```
Comment (pamapa) [inline src/SessionMonitor.ts:156]: i do not understand why this should work:
- We still use session.sub at line 152
- With your changes here i would expect that `_raiseUserSessionChanged` is called all the time when the user is still ok...

Comment (pamapa) [reply src/SessionMonitor.ts:156]: lets keep the `sub` claim

**Review by pamapa — APPROVED** [review]

**Comment by pamapa** [comment]
CI pipline fails with:
```
You have changed the public API signature for this project. Please copy the file "temp/oidc-client-ts.api.md" to "docs/oidc-client-ts.api.md", or perform a local build (which does this automatically). See the Git repo documentation for more info.
```

and with
```
npm run lint

> oidc-client-ts@2.4.0 lint
> eslint --max-warnings=0 --cache .

/home/runner/work/oidc-client-ts/oidc-client-ts/src/SessionMonitor.ts
Error:   57:36  error  Unexpected separator (;)  @typescript-eslint/member-delimiter-style
```

please resolve

**Comment by pamapa** [comment]
I just found that `sid` is also part of `SessionStatus` returned in `querySessionStatus`. We should either remove it form there as well or as an alternative we make `sid` optional: If it is defined consider it in the `SessionMonitor` else ignore it. What do you think about the alternative?

**Comment by bachratyg** [comment]
I'd rather not put back a partial `sid` check. That would mean the lib sometimes decides for me if there's a change, and sometimes not. It doesn't feel consistent. This is opinionated behavior, it should be more explicit.

Alternatively the `SessionState` could be extended to include the full user. `querySessionStatus` already does a reduced silent sign-in (only the `openid` scope is requested), might as well do the full thing. The user can also be added to the events then the subscriber can do as they see fit e.g.
```
userManager.events.addUserSessionChanged(async newUser => {
    if( newUser.sid !== oldUser.sid ) {
        // previous handler logic
    }
});
```

**Comment by pamapa** [comment]
> I'd either remove the `sid` or alternatively extend `SessionState` to include the full user.

~~if we are going with the `User`, we could drop `SessionState` and directly use `User`:
-> `querySessionStatus(args: QuerySessionStatusArgs = {}): Promise<User| null>`~~
The returned object would contain the currently signed-in user, would it be useful to also provide the previous signed-in user?

The main question is how do we then handle the types for `monitorAnonymousSession` feature. Do you know the use-case for this feature? If it is still useful, we should add some documentation to the settings property else drop it?

Probably something we already have:
```ts
const tmpUser = {
  session_state: session.session_state,
  profile: session.sub /*&& session.sid*/ ? {
    sub: session.sub,
    //sid: session.sid,
  } : null,
};
```
This would also simplify the type for the parameter of `SessionMonitor._start`.

**Comment by bachratyg** [comment]
> ~if we are going with the `User`, we could drop `SessionState` and directly use `User`: -> `querySessionStatus(args: QuerySessionStatusArgs = {}): Promise<User| null>`~

The `session_state` is still needed for monitoring the anonymous session. It should be exposed separately from `User`.

> The returned object would contain the currently signed-in user, would it be useful to also provide the previous signed-in user?

Subscriber can always call `getUser` but it would certainly be convenient.

> The main question is how do we then handle the types for `monitorAnonymousSession` feature. Do you know the use-case for this feature? If it is still useful, we should add some documentation to the settings property else drop it?

It could be useful for apps that have public features e.g. some sites show a popup when you sign in on another page. The only problem I see here is that a new notification could be triggered while the previous handler is still running but that is already the case with `UserSessionChanged`.

**Comment by pamapa** [comment]
> I'd either remove the `sid` or alternatively extend `SessionState` to include the full user

so lets remove that too...

**Comment by bachratyg** [comment]
Removed `sid` for now. I will need a bit more time to improve querySessionStatus with the full profile

**Review by pamapa — APPROVED** [review]

**Comment by pamapa** [comment]
thanks for contributing

**Pull request merged**